File: Java/LinearSearch.java
import java.util.Scanner;

class LinearSearch {
    public static int linearSearch(int[] arr, int target) {
        for(int i = 0; i < arr.length; i++) {
            if(arr[i] == target) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter your array size: ");
        int maxSize = sc.nextInt();

        int[] arr = new int[maxSize];

        System.out.println("enter array elements: ");
        for(int i = 0; i < maxSize; i++) {
            arr[i] = sc.nextInt();
        }
        
        System.out.println();

        for(int a = 0; a < arr.length; a++) {
            System.out.print(arr[a] + "  ");
        }

        System.out.println();

        System.out.println("enter number to search: ");
        int target = sc.nextInt();

        int result = linearSearch(arr, target);

        if(result == -1) {
            System.out.println("no such element found.");
        } else {
            System.out.println("element at index: " + result);
        }
    }
}

File: C++/LinearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int arr[], int size, int target) {
    for (int i = 0; i < size; i++) {
        if (arr[i] == target) {
            return i;
        }
    }
    return -1;
}

int main() {
    int maxSize;

    cout << "Enter your array size: " << endl;
    cin >> maxSize;

    int arr[maxSize];

    cout << "enter array elements: " << endl;
    for (int i = 0; i < maxSize; i++) {
        cin >> arr[i];
    }

    cout << endl;

    for (int a = 0; a < maxSize; a++) {
        cout << arr[a] << "  ";
    }

    cout << endl;

    int target;
    cout << "enter number to search: " << endl;
    cin >> target;

    int result = linearSearch(arr, maxSize, target);

    if (result == -1) {
        cout << "no such element found." << endl;
    } else {
        cout << "element at index: " << result << endl;
    }

    return 0;
}